go/types/union.go

```diff
@@ -43,8 +41,32 @@ func makeSimplifiedUnion(in ...*Type) *Type {
 		ts[t] = struct{}{}
 	}
 
-	// makeSimplifiedUnionImpl de-cycles internally.
-	return makeSimplifiedUnionImpl(ts)
+	// makeUnionImpl de-cycles internally.
+	return makeUnionImpl(ts, true)
+}
+
+// makeUnion returns a type that is the union of all the input types.
+//
+// The resulting type is similar to the type produced by makeSimplifiedUnion
+// with the one exception: Each named Struct will contain the union of all fields in
+// the inputs as opposed to just the common fields. In effect this means the resulting
```

Comment:
It's not true that the resulting type is a subtype:

`makeUnion(List<A>, List<B>) => List<A|B>`.

`List<A|B>` is a supertype of `List<A>` and `List<B>`, not a subtype.